python/semantic_kernel/orchestration/sk_function_base.py

```diff
@@ -85,6 +85,14 @@ def is_native(self) -> bool:
         """
         pass
 
+    @property
+    @abstractmethod
+    def is_function_call(self) -> bool:
+        """
+        Whether the function is a function call that is passed to the API.
```

Comment:
Semi-called it out earlier, but the functions API is really specific to OpenAI. I think it would be better to have a method in the base class called "serialize" or similar, and then pass a serialization function to that method. It's useful to be able to serialize the calling pattern of an SK function, but the structure of the data may need to differ depending on how it's used (ex: OpenAI functions API, Planner, etc.)

Reply:
yeah, serialize or something similar makes sense, I'll have a look